mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -697,3 +698,66 @@ def __repr__(self):
         repr_str += '(bbox_enlarge_range={})'.format(
             self.bbox_enlarge_range.tolist())
         return repr_str
+
+
+@PIPELINES.register_module()
+class SweepPointSample(object):
+    """Sweep point voxel sample.
+
+    Apply voxel sampling to multiple sweep points.
+
+    Args:
+        name (str): Name of the dataset.
+        num_points (int): Number of points to be sampled.
+    """
+
+    def __init__(self, cur_sweep_cfg, prev_sweep_cfg):
+        self.cur_sweep_sampler = VoxelGenerator(**cur_sweep_cfg)
+        self.prev_sweep_sampler = VoxelGenerator(**prev_sweep_cfg)
+
+    def __call__(self, results):
+        """Call function to sample points from multiple sweeps.
+
+        Args:
+            input_dict (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Results after sampling, 'points', 'pts_instance_mask' \
+                and 'pts_semantic_mask' keys are updated in the result dict.
+        """
+        points = results['points']
+        cur_points_num = results['cur_points_num']
+        cur_sweep_points = points[:cur_points_num]
+        prev_sweeps_points = points[cur_points_num:]
+
+        voxels, coors, num_points_per_voxel = self.cur_sweep_sampler.generate(
+            cur_sweep_points)
+        max_voxels = self.cur_sweep_sampler._max_voxels
+        if voxels.shape[0] >= max_voxels:
```

Comment:
This should already be done in `self.cur_sweep_sampler` [here](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/voxel/voxel_generator.py#L185). Please check the code.